base/traits.jl

```diff
@@ -20,3 +20,40 @@ TypeArithmetic(instance) = TypeArithmetic(typeof(instance))
 TypeArithmetic(::Type{<:AbstractFloat}) = ArithmeticRounds()
 TypeArithmetic(::Type{<:Integer}) = ArithmeticOverflows()
 TypeArithmetic(::Type{<:Any}) = ArithmeticUnknown()
+
+# trait for objects that support ranges with regular step
+"""
+    TypeRangeStep(instance)
+    TypeRangeStep(T::Type)
```

Comment:
these aren't exported, but should they go in the stdlib? are the other traits in this file there?